tests/31sync/06state.pl

```diff
@@ -338,6 +338,64 @@
    };
 
 
+test "A change to displayname should not result in a full state sync",
+   requires => [qw( first_api_client can_sync )],
+   bug => 'SYN-515',
+
+   check => sub {
+      my ( $http ) = @_;
+
+      my ( $user, $filter_id, $room_id, $next_batch );
+
+      my $filter = {
+         room => {
+            timeline  => { types => [] },
+            state     => { types => [ "a.madeup.test.state" ] },
+            ephemeral => { types => [] },
+         },
+         presence => {types => [] },
+      };
+
+      matrix_register_user_with_filter( $http, $filter )->then( sub {
+         ( $user, $filter_id ) = @_;
+
+         matrix_create_room( $user );
+      })->then( sub {
+         ( $room_id ) = @_;
+
+         matrix_put_room_state( $user, $room_id,
+            type      => "a.madeup.test.state",
+            content   => { "my_key" => 1 },
+            state_key => ""
+         );
+      })->then( sub {
+         matrix_sync( $user, filter => $filter_id );
+      })->then( sub {
+         my ( $body ) = @_;
+
+         $next_batch = $body->{next_batch};
```

Comment:
You're setting `$next_batch` but then don't appear to be actually using it afterwards